examples/tutorials/inset.py

```diff
@@ -0,0 +1,116 @@
+"""
+Adding an inset to the figure
+======================
+
+To plot an inset figure inside another larger figure, we can use the
+:meth:`pygmt.Figure.inset` method. After a large figure has been created,
+call ``inset`` using a ``with`` statement, and new plot elements will be
+added to the inset figure instead of the larger figure.
+"""
+
+import pygmt
+
+########################################################################################
+#
+# Prior to creating an inset figure, a larger figure must first be plotted. In the
+# example below, :meth:`pygmt.Figure.coast` is used to create a map of the US state of
+# Massachusetts.
+
+fig = pygmt.Figure()
+fig.coast(
+    # Sets the region plotted in the larger figure
+    region=[-74, -69.5, 41, 43],
+    # Sets the state boundaries to be plotted with thin lines
+    borders="2/thin",
+    # Plots all shorelines with thin lines
+    shorelines="thin",
+    # Sets a 15 centimeter figure showing a Mercator projection
+    projection="M15c",
+    # Sets the color of the land to "lightyellow"
+    land="lightyellow",
+    # Sets the color of the water to "lightblue"
+    water="lightblue",
+    # Sets the frame to use the automatic settings
+    frame="a",
```

Comment:
```suggestion
    region=[-74, -69.5, 41, 43],  # Set bounding box of the large figure
    borders="2/thin",  # Plot state boundaries with thin lines
    shorelines="thin",  # Plot coastline with thin lines
    projection="M15c",  # Set Mercator projection and size of 15 centimeter
    land="lightyellow",  # Color land areas light yellow
    water="lightblue",  # Color water areas light blue
    frame="a",  # Set frame with annotation and major tick spacing
```
Keep things inline. And the `frame="a"` isn't actually "auto", see https://docs.generic-mapping-tools.org/latest/gmt.html#b-full :wink: